apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloPlugin.kt

```diff
@@ -157,6 +88,23 @@ open class ApolloPlugin : Plugin<Project> {
             .withFallback(project.objects, compilationUnit.service)
             .withFallback(project.objects, compilationUnit.apolloExtension)
 
+        val generateKotlinModels = compilerParams.generateKotlinModels.getOrElse(false)
+
+        /**
+         * To avoid project.afterEvaluate, all tasks are registered before we know if the user configured
+         * services or set generateKotlinModels. Here we disable the ones that do not make sense once we
+         * know the actual user configuration.
+         */
+        var enabled = true
+        if (generateKotlinModels != compilationUnit.kotlin) {
```

Comment:
This could be nicer with a when statement. Right now I'm on mobile. I'm thinking something like this 

```
it.enabled = when {
    condition -> false
    else -> true
}
```

Reply:
yup, much nicer!